### PR TITLE
bump rust-ncbitaxonomy to 1.0.5

### PR DESCRIPTION
**Pull request by pvanheus**
update rust-ncbitaxonomy to version 1.0.5

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by pvanheus** [comment]
@BiocondaBot please add label

**Comment by pvanheus** [comment]
A quick note on build: the build.sh now uses LD_LIBRARY_PATH to pass in the location of the sqlite library. I found that this, rather than other environment variables, leads to a successful build.

**Comment by mbargull** [comment]
@BiocondaBot please fetch artifacts

File: recipes/rust-ncbitaxonomy/build.sh [hunk 1]
```diff
@@ -11,4 +11,4 @@ fi
 LD_LIBRARY_PATH="$PREFIX/lib"
 export LD_LIBRARY_PATH
 # build statically linked binary with Rust
-RUST_BACKTRACE=1 C_INCLUDE_PATH=$PREFIX/include cargo install --verbose --root $PREFIX --path .
+RUST_BACKTRACE=1 C_INCLUDE_PATH=$PREFIX/include cargo install --no-track --verbose --root $PREFIX --path .
```
Comment (mbargull) [inline recipes/rust-ncbitaxonomy/build.sh]: Without `--no-track` this created some
```
.crates2.json
.crates.toml
```
files.

Comment (pvanheus) [reply recipes/rust-ncbitaxonomy/build.sh]: ok thanks, I didn't know that.

Comment (mbargull) [reply recipes/rust-ncbitaxonomy/build.sh]: Me neither, just noticed those surplus files and `--no-track` is supposed to not create them AFAIU.

**Comment by mbargull** [comment]
@BiocondaBot please fetch artifacts

**Comment by mbargull** [comment]
@BiocondaBot please merge

**Pull request merged**